composer/utils/checkpoint.py

```diff
@@ -256,17 +277,50 @@ def _download_checkpoint(
     return composer_states_filepath, extracted_checkpoint_folder, extracted_rank_n
 
 
+def glob_filter(exclude_globs: List[str]) -> Callable[[Dict], None]:
+    """Provides a function which deletes all subparts of a dictionary based on a list of paths."""
+
+    def filter_func(state_dict: Dict) -> None:
+
+        def recur_filter_func(state_dict: Dict, exclude_glob: List[str]):
+            # End of path, delete key in state_dict
+            if len(exclude_glob) == 1:
+                del state_dict[exclude_glob[0]]
+            # Wildcard, descend down all keys
+            elif exclude_glob[0] == "*":
+                for key in state_dict.keys():
+                    recur_filter_func(state_dict[key], exclude_glob[1:])
+            # Step down through path
+            else:
+                recur_filter_func(state_dict[exclude_glob[0]], exclude_glob[1:])
```

Comment:
If the `state_dict` does not have this key, then I think it's raise a hard-to-debug error (e.g. `KeyError 'bias' not found`). Could we make it more debuggable by re-raising the exception to include the full glob path that caused the error to be generated (e.g. `state/model/` has no key `conv1.bias`)

Another thought -- the implementation would raise an exception if one specifies an exclude glob that is not in the state dict. Should this be a warning or a silent ignore (for better portability?)

Reply:
Will rework to log ignored keys -- `log.info` for ones excluded and `log.warn` for passed in globs that don't match